### PR TITLE
[Docs] Added the guideline for documentation

### PR DESCRIPTION
**Pull request by pedrominatel**
## Summary

Added the guideline for the documentation.

## Impact

None.

## Related links

None.

### CODE REVIEW

File: docs/source/guides/docs_contributing.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+The guideline works to give you the directions and to keep the documentation more concise, helping users to better understand the structure.
+
+About Documentation
+-------------------
+
+We all know how important documentation is. This project is no different.
+
+This documentation was created in a collaborative and open way, letting everyone contribute, from a small typo fix to a new chapter writing. We try to motivate our community by giving all the support needed through this guide.
+
+The documentation is in **English only**. Future translations can be added since we finish the essential content in English first.
```
Comment (me-no-dev) [inline docs/source/guides/docs_contributing.rst:19]: `since` -> `when`

File: docs/source/guides/docs_contributing.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+Sections
+--------
+
+The Arduino ESP32 is structured in some sections to make it easier to maintain. Here is a brief description of this structure.
+
+API
+***
+
+In this section, you will include all the documentation about drivers, libraries and any other related to the core.
```
Comment (me-no-dev) [inline docs/source/guides/docs_contributing.rst:117]: make it a bit more clear, that only API documentation should go there, not general info or help/FAQ on the drivers/libs

File: docs/source/guides/docs_contributing.rst (hunk 1 above)
Comment (pedrominatel) [reply docs/source/guides/docs_contributing.rst]: Done!

File: docs/source/guides/docs_contributing.rst [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+Guides
+******
+
+This is the place to add the guides for common applications, IDEs configuration, and any other information that can be used as a guideline.
+
+Tutorials
+*********
+
+If you want to add a specific tutorial related to the Arduino for ESP32, this is the place. The intention is not to create a blog or a demo area, but this can be used to add some complex description or to add some more information about APIs.
```
Comment (VojtechBartoska) [inline docs/source/guides/docs_contributing.rst:140]: * Arduino **core** for ESP32

File: docs/source/guides/docs_contributing.rst [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+.. code-block::
+
+    .. code-block:: arduino
+        bool begin(); //Code example
+
+Links
+*****
+
+To include links to external content, you can use two ways. The first
```
Comment (VojtechBartoska) [inline docs/source/guides/docs_contributing.rst:287]: MAybe do here separete header? As "First option" or "Second option"

**Review by VojtechBartoska — APPROVED** [review]
Looks very good, I left few small comments.

**Comment by pedrominatel** [comment]
PTAL @VojtechBartoska.

**Comment by VojtechBartoska** [comment]
@pedrominatel You found even more things, hehe. LGTM, ready to be merged.

**Pull request merged**